web/controllers/friends_controller.ex

```diff
@@ -5,34 +5,54 @@ defmodule Entice.Web.FriendsController do
 
   plug :ensure_login
 
+  #Case to think about:
+  #Player A adds friend Player B under name N
+  #Player B deletes account (and so all chars are deleted)
+  #Player C creates acccount and char under name N
+  #Player A adds Player C under name N
+  #Now in friends DB: (N, PlayerA.id, PlayerB.id) and (N, PlayerA.id, PlayerC.id)
+  #Since all queries are done by name it'll be an issue.
+  #Solutions:
+  #Delete all friends with PlayerB.id in them
+
+
+  #TODO: add map once it's server sided, order by creation date
   @doc "Returns all friends of connected account."
   def index(conn, _params) do
     id = get_session(conn, :client_id)
     {:ok, friends} = Entice.Web.Client.get_friends(id)
 
-    friends = for friend <- friends, do: friend.friend_account.id
+    results = []
+    results = for friend <- friends do
+      {:ok, status, name} = get_status(friend.base_name)
+      map = %{base_name: friend.base_name, current_name: name, status: status}
+      results = results ++ [map]
```

Comment:
I was unsure if 'variable = for ...' in elixir worked like list comprehension or if variable was set to the last iteration's return. If I understood what you said correctly, it's the former. I'll make the necessary changes =).